src/lib.rs

```diff
@@ -225,40 +224,56 @@ impl PolicyServer {
             router = Router::new().merge(router).merge(pprof_router);
         }
 
+        let readiness_probe_router = Router::new().route("/readiness", get(readiness_handler));
+
         Ok(Self {
             router,
+            readiness_probe_router,
             callback_handler,
             callback_handler_shutdown_channel_tx,
             addr: config.addr,
             tls_config,
+            readiness_probe_addr: config.readiness_probe_addr,
         })
     }
 
     pub async fn run(self) -> Result<()> {
-        // Start the CallbackHandler
+        let notify = Notify::new();
+
         let mut callback_handler = self.callback_handler;
         let callback_handler = tokio::spawn(async move {
             info!(status = "init", "CallbackHandler task");
             callback_handler.loop_eval().await;
             info!(status = "exit", "CallbackHandler task");
         });
 
-        if let Some(tls_config) = self.tls_config {
-            axum_server::bind_rustls(self.addr, tls_config)
-                .serve(self.router.into_make_service())
-                .await?;
-        } else {
-            axum_server::bind(self.addr)
-                .serve(self.router.into_make_service())
-                .await?;
+        let api_server = async {
+            if let Some(tls_config) = self.tls_config {
+                let server_with_tls = axum_server::bind_rustls(self.addr, tls_config);
+                notify.notify_one();
+
+                server_with_tls.serve(self.router.into_make_service()).await
+            } else {
+                let server = axum_server::bind(self.addr);
+                notify.notify_one();
+
+                server.serve(self.router.into_make_service()).await
+            }
+        };
+
+        let readiness_probe_server = async {
+            notify.notified().await;
+
+            axum_server::bind(self.readiness_probe_addr)
```

Comment:
I would add some logging telling the user the `/readiness` probe is ready and listening over port XYZ